packages/babel-parser/src/parser/util.js

```diff
@@ -147,4 +147,22 @@ export default class UtilParser extends Tokenizer {
       );
     }
   }
+
+  checkYieldAwaitInDefaultParams() {
+    if (
+      this.state.yieldPos &&
+      (!this.state.awaitPos || this.state.yieldPos < this.state.awaitPos)
```

Comment:
Can't we avoid this check by keeping the two state properties together?

Reply:
How would you put them together without loosing the information about if it is either await or yield?

Reply:
You are right, I didn't thought about it :+1: